docs/advanced-topics/how-it-works.md

```diff
@@ -6,3 +6,201 @@ description: Low level details about the framework
 
 Component-lifecycle state machine, vdom diff algorithm.
 
+## Under the hood of the `html!` macro
+
+Rust's macros are an example of "metaprogramming" - the term is the Greek "meta" combined with 
+"programming", although the meaning of "meta" in this context is not the Greek one but rather its
+meaning in epistemology (the study of knowledge) which means about. Metaprogramming is "programming"
+about programming. In this context this is about computer programs which write other computer
+programs though metaprogramming doesn't have to explicitly be about writing programs.
```

Comment:
Perhaps not – I personally like reading about bits of trivia in programming books so that's why I put it in. If the consensus is that it'd be better without this I'm happy to remove it.

Reply:
I really don't mind the random trivia either, so I'm fine with leaving it in.
What do you think about adding my proposed section about macros though?
Also, how about we mark the section as trivia? Even something as simple as using a quote should do the trick:
> **_TRIVIA:_** Rust's macros are an example of ...

We could also use inline HTML if we wanted to be all fancy.
What do you think?

Reply:
I like the idea of marking it as trivia. Your proposed section sounds good – I'll add it. I feel our documentation has to be useable by both those who've come across Yew as Javascript programmers as well as those who are coming at it from a Rsut angle which makes it a delicate balance between adding too much introductory stuff which a Rust programmer would already know versus leaving Javascript developers seriously confused.